### ISSUE TITLE
user.present:groups vs group.present:members

### ISSUE THREAD

**Issue by kkaig** [saltstack/salt#28706]
Hi, Salt has an option in group.present to set a list of members, but this conflicts with the user.present:groups list. If you don't pass a list of groups in user.present, salt assumes the empty list, and (as per the docs) it will remove the user from all groups.

The following code will work, but will always report a change, since it removes myuser from all groups, then adds it back to the deploy group:
```salt
myuser:
  user.present:
    - shell: /bin/bash
deploy:
  group.present:
    - members:
      - myuser
```
The correct behaviour would be for salt not to change the user's groups when running user.present unless the groups: [] list is explicitly passed.

Thanks

**Comment by jfindlay** [comment]
@kkaig, thanks for the report.

**Comment by rallytime** [comment]
@kkaig Does the fix in #30824 help resolve this issue for you?
